# Statistics Adv-4

Q1: What is the difference between a t-test and a z-test? Provide an example scenario where you would
use each type of test.

Ans. t-test and z-test are two types of tests used in Hypothesis testing derive conclusions whether the null hypothesis is true or not. 

- **z-test** is used when the population standard deviation is given. Also the sample size should be large. In general, $n \ge 30 $ is the limit for z-test
- **t-test** is used when the population standard deviation is not given but the sample standard deviation is given. It can also be used when sample size is very less.

Q2: Differentiate between one-tailed and two-tailed tests.

Ans. One-tailed and two-tailed tests are two types of tests used in Hypothesis testing. 

- **One tailed tests**: Here we test the hypothesis about any parameter only in one direction. Critical region is defined on one side of region. For example, if the null hypothesis is that average IQ is 100 and we have to test that if some new drug improves the IQ or not, then we we are doing a one tailed test:
    - $H_0 : \mu=100$ : The average IQ is 100
    - $H_1 : \mu \gt 100$ : The average IQ is more than 100


- **Two tailed tests**: Here we test the hypothesis about any parameter in both directions, we see the increase as well as decrease. The critical region is defined on both sides of region. Here we see if there is any significant change in value. If we continue with above example, then a two tailed test in that case will be when we are just testing if there is any effect of drug on IQ, it may be positive or negative, IQ may improve or decrease.
    - $H_0 : \mu=100$ : The average IQ is 100
    - $H_1 : \mu \ne 100$ : The average IQ is not equal to 100

Q3: Explain the concept of Type 1 and Type 2 errors in hypothesis testing. Provide an example scenario for
each type of error.

Ans. In Hypothesis testing an error occurs when we make a wrong decision based on data. Type 1 and Type 2 errors are two types of errors that can occur in hypothesis testing:

- **Type 1 error**: The type 1 error occurs when we reject the Null Hypothesis even when it is true in actual case. It is also known as false positive.

    Example Scenario for Type 1 Error:
    Imagine a clinical trial testing a new drug for a disease. The null hypothesis ($H_0$) states that the drug has no effect. However, if researchers incorrectly reject and conclude that the drug is effective, it could lead to the drug being prescribed widely when it doesn't actually work.

- **Type 2 error**: The type 2 error occurs when we accept the Null Hypothesis even when it is false in actual case. It is also known as false negative.

    Example Scenario for Type 2 Error:
    Continuing from the previous example, let's assume the new drug is indeed effective. However, due to limited sample size or other factors, the clinical trial fails to show a statistically significant difference between the treated and control groups. If researchers fail to reject ($H_0$) and conclude that the drug is not effective, patients might miss out on a potentially beneficial treatment.


Q4: Explain Bayes's theorem with an example.

Ans. Bayes's theorem is used to give probability of an event when a given event has already occured:

For two events, Bayes theorem is given by:
$$ P\left(A|B\right) = \frac{P\left(A\right).P\left(B|A\right)}{P\left(B\right)}$$

where:
- $ P\left(A|B\right)$: Probability of occurenc of A given B is true
- $ P\left(B|A\right)$: Probability of occurenc of B given A is true
- $ P(A) $: Probability of occurenc of A 
- $ P(B) $: Probability of occurenc of B

Example: Suppose a box has 3 red balls and 2 blue balls. Then:
- The probability of picking a red ball $P(R) = \frac{3}{5}$
- The probability of picking a blue ball $P(B) = \frac{2}{5}$
- The probability of picking a red ball if a blue ball is picked $P(R|B) = \frac{3}{4}$

Now using Bayes theorem, the probability of picking a blue ball if a red ball is picked:

$$ P\left(B|R\right) = \frac{P\left(B\right).P\left(R|B\right)}{P\left(R\right)}$$

$$ P(B|R) = \frac{2/5 * 3/4}{3/5} $$

$$ P(B|R) = \frac{6/20}{3/5} $$

$$ P(B|R) = \frac{1}{2} $$

$$ P(B|R) = 0.5 $$

Q5: What is a confidence interval? How to calculate the confidence interval, explain with an example.

Ans. A confidence interval is a range of values which is derived in interval estimate. We say that the population parameter is likely to fall into the given range of values with a certain level of confidence. For example a 95% confidence interval for populaiton mean means that there are 95% chance that the population mean lies in that given range of values.

We can use the given formaula to find the confidence interval of population mean:
$$CI = \bar{x} \pm Z_{\alpha/2}. \left(\frac{\sigma}{\sqrt{n}}\right)$$

For example:  We have $n$=50, $\bar{x} = 4$, $\sigma=1.5$, and $\alpha=1-0.99=0.01$

Solution using pyhton is as follows:

In [1]:
from math import sqrt
import scipy.stats as stat

std = 1.5
sample_mean = 4
n = 50
confidence = 99
alpha = 1 - confidence/100

z = abs(stat.norm.ppf(alpha/2))
margin = z*(std/sqrt(n))
CI = (sample_mean-margin, sample_mean+margin)
print(f"The {confidence}% confidence interval of population mean is {CI}")

The 99% confidence interval of population mean is (3.4535840896844654, 4.546415910315535)


Q6. Use Bayes' Theorem to calculate the probability of an event occurring given prior knowledge of the
event's probability and new evidence. Provide a sample problem and solution.

Ans. For two events, Bayes theorem is given by:
$$ P\left(A|B\right) = \frac{P\left(A\right).P\left(B|A\right)}{P\left(B\right)}$$

Suppose there's a drug test for a certain performance-enhancing drug used by athletes. Let's define the following:

- Event A: The athlete is using the performance-enhancing drug.
- Event B: The athlete tests positive for the drug.

We know the following probabilities:

The probability that an athlete is using the drug (prior probability): $P(A) = 0.05 (5%)$
The probability that an athlete tests positive given that they are using the drug: $P(B|A) = 0.98 (98%)$
The probability that an athlete tests positive given that they are not using the drug: $P(B|\bar{A}) = 0.04 (4%)$
Now, we want to find the probability that an athlete is actually using the drug given that they test positive, i.e., $P(A|B)$

$$P(B)=P(B∣A)⋅P(A)+P(B∣\bar{A})⋅P(\bar{A})$$
$$P(B)= 0.98⋅0.05 + 0.04⋅0.95$$
$$P(B)= 0.049 + 0.038 = 0.087$$

Now, using Bayes' Theorem:
$$ P\left(A|B\right) = \frac{P\left(A\right).P\left(B|A\right)}{P\left(B\right)}$$
$$ P\left(A|B\right) = \frac{0.98⋅0.05}{0.087}$$
$$P(A∣B)≈0.5632$$

So, even though the athlete tests positive for the drug, the probability that they are actually using the drug is about 56.32%. This demonstrates how Bayes' Theorem allows us to update our beliefs about the probability of an event based on new evidence.

Q7. Calculate the 95% confidence interval for a sample of data with a mean of 50 and a standard deviation
of 5. Interpret the results.

Ans. The Sample size is not given in question, let us assumne a sample size of 100, as population standard deviation is not given only sample standard deviationis given, so we use t statistics.

$n=100$, $\bar{x}=50$, $S=5$ confidence interval, $CI=95$, so we have $\alpha=1-0.95=0.05$

We can use the given formaula to find the confidence interval of population mean:
$$CI = \bar{x} \pm t_{\alpha/2}. \left(\frac{S}{\sqrt{n}}\right)$$

In [38]:
from math import sqrt
import scipy.stats as stats

sample_mean = 50
std = 5
n = 100
confidence = 95
alpha = 1 - confidence/100

t = stats.t.isf(alpha/2, n-1)
margin = (t*std)/sqrt(n)

CI = (sample_mean-margin, sample_mean+margin)
print(f"The {confidence}% confidence interval of population mean is {CI}")

The 95% confidence interval of population mean is (49.00789152424566, 50.99210847575434)


This means that there are 95% chances that the population mean lies between 49.008 and 50.992

Q8. What is the margin of error in a confidence interval? How does sample size affect the margin of error?
Provide an example of a scenario where a larger sample size would result in a smaller margin of error.

Ans. The margin of error (MOE) is a measure of the uncertainty associated with estimating a population parameter using a sample. In the context of a confidence interval, the margin of error indicates the range within which the true population parameter is likely to fall, given a certain level of confidence.

The margin of error is often expressed as a positive value, and it represents half of the width of the confidence interval. The MOE is the value that we add and subtract from the point estimate to get the upper and lower bounds of the interval.

The formula to calculate the margin of error for a population mean, when the population standard deviation is known, is:
$$ MOE = Z_{\alpha/2}. \left(\frac{\sigma}{\sqrt{n}}\right)$$

In case populaltion proportion is given, then we use
$$ MOE = Z_{\alpha/2}. \sqrt{\frac{p.(1-p)}{n}}$$

**Effect of sample size**- The margin of error is inversely proportional to the square root of sample size. So if we increase the sample size, margin of error reduces. If we decrease the sample size, margin of error increases.

For example, if we have $\sigma=10$, then let us calculate the margin of error for two sample size, $n_1 = 50$ and $n_2 = 5000$ with a $95\%$ confidence level using python:

In [11]:
from math import sqrt
import scipy.stats as stat

std = 10
n1 = 50
n2 = 5000
confidence = 90
alpha = 1 - confidence/100

z = abs(stat.norm.ppf(alpha/2))
margin_1 = z*(std/sqrt(n1))
margin_2 = z*(std/sqrt(n2))

print(f"The margin of error for sample size {n1} is {margin_1}")
print(f"The margin of error for sample size {n2} is {margin_2}")


The margin of error for sample size 50 is 2.326174307353348
The margin of error for sample size 5000 is 0.23261743073533483


Thus we see that a larger sample size results in a smaller margin of error.

Q9. Calculate the z-score for a data point with a value of 75, a population mean of 70, and a population
standard deviation of 5. Interpret the results.

Ans. Given: $\mu=70$, $x=75$, $\sigma=5$

So, the Z-score for $x$ is :
$$ Z = \frac{x - \mu}{\sigma} $$

$$ Z = \frac{75 - 70}{5} $$

$$ Z = \frac{5}{5} $$

$$ \fbox{Z = 1} $$

Interpretation of result: Z=+1 means that the point x=75 is 1 standard deviation away from the mean on right side.

Q10. In a study of the effectiveness of a new weight loss drug, a sample of 50 participants lost an average
of 6 pounds with a standard deviation of 2.5 pounds. Conduct a hypothesis test to determine if the drug is
significantly effective at a 95% confidence level using a t-test.

Ans. We have $\bar{x}=6$, $n=50$, $S=2.5$, $\alpha=0.05$. The null and alternative hypothesis are:
- $H_0$: $\mu=0$ : this means mean weight loss is 0 and the drug is not effective
- $H_1$: $\mu \ne 0$ : this means the drug is effective

As population standard deviation is not given, we will use t-statistics. Using Pyhton, the solution can be obtained as shown:

In [2]:
import scipy.stats as stat
import numpy as np

pop_mean = 0
sample_mean = 6
std = 2.5
significance_level = 0.05
sample_size = 50
dof =sample_size-1

t = (sample_mean-pop_mean)/(std/np.sqrt(sample_size))
p = stat.t.ppf(1-significance_level/2, dof)

print("RESULT:")

if t<-p or t>p:
    print("We reject the Null Hypothesis. The new weight loss drug is significantly effective")
else:
    print("We accept the Null Hypothesis. The new weight loss drug is not significantly effective")


RESULT:
We reject the Null Hypothesis. The new weight loss drug is significantly effective


Q11. In a survey of 500 people, 65% reported being satisfied with their current job. Calculate the 95%
confidence interval for the true proportion of people who are satisfied with their job.

Ans. For a proportion, the formula for confidence interval is as follows:
$$CI = p \pm Z_{\alpha/2}. \sqrt{\frac{p.(1-p)}{n}}$$

where:
- $p$ is the proportion
- $Z_{\alpha/2}$ is the z score
- $n$ is the sample size

Solution using pyhton is as shown:

In [8]:
from math import sqrt
import scipy.stats as stat

p = 65/100
n = 500
confidence = 95
alpha = 1 - confidence/100

z = abs(stat.norm.ppf(alpha/2))
margin = z*(sqrt(p*(1-p)/n))
CI = (sample_mean-margin, sample_mean+margin)

print(f"""The {confidence}% confidence interval  for the true proportion of people who are satisfied with their job is 
{CI}""")

The 95% confidence interval  for the true proportion of people who are satisfied with their job is 
(3.958192539380921, 4.041807460619079)


Q12. A researcher is testing the effectiveness of two different teaching methods on student performance.
Sample A has a mean score of 85 with a standard deviation of 6, while sample B has a mean score of 82
with a standard deviation of 5. Conduct a hypothesis test to determine if the two teaching methods have a
significant difference in student performance using a t-test with a significance level of 0.01.

Ans. The size of smaple is not given in question. Let us assume that both samples have same size $n=50$.The formula used in this scenario will be:

$$ t = \frac{\bar{x}_1 - \bar{x}_2}{s_p.\sqrt{\frac{2}{n}}} $$

where $s_p$ is the pooled standard deviation
$$ s_p = \sqrt{\frac{s_1^2 + s_2^2}{2}}$$

- $s_1 = 6$ : standard deviation of first sample
- $s_2 = 5$ : standard deviation of second sample
- $\bar{x}_1 = 85$ : mean of first sample 
- $\bar{x}_2 = 82$ : mean of second sample 
- $n = 50$ : sample size

The null and alternative hypothesis are:
- $H_0$: $\bar{x}_1 - \bar{x}_2=0$ : the means are different, the two teaching methods don'ta significant difference in student performance 
- $H_1$: $\bar{x}_1 - \bar{x}_2 \ne 0$ : the means are not different, the two teaching methods have a significant difference in student performance 

Solution using python is as shown:

In [30]:
import scipy.stats as stat
import numpy as np

mean1 = 85
mean2 = 82
s1 = 6
s2 = 5
significance_value = 0.01
size = 50
dof = 2*(size-1)

std_pooled = np.sqrt((s1**2 + s2**2)/2)
t = (mean1-mean2)/(std_pooled*np.sqrt(2/size))
critical = stat.t.ppf(1-significance_value/2, dof)
    
print(f"t : {t}")
print(f"critical value : {critical}")
if t>critical or t<-critical:
    print("We reject the Null Hypothesis. The two teaching methods have a significant difference in student performance ")
else:
    print("We fail to reject the Null Hypothesis. The two teaching methods don't have a significant difference in student performance ")
    

t : 2.716072381275556
critical value : 2.626931094814024
We reject the Null Hypothesis. The two teaching methods have a significant difference in student performance 


Q13. A population has a mean of 60 and a standard deviation of 8. A sample of 50 observations has a mean
of 65. Calculate the 90% confidence interval for the true population mean.

Ans. $\mu=60$, $\sigma=8$, $n=50$, $\bar{x}=65$, confidence interval, $CI=90$, so we have $\alpha=1-0.90=0.1$

We can use the given formaula to find the confidence interval of population mean:
$$CI = \bar{x} \pm Z_{\alpha/2}. \left(\frac{\sigma}{\sqrt{n}}\right)$$

Solution using python is as follows:

In [4]:
from math import sqrt
import scipy.stats as stat

pop_mean = 60
std = 8
sample_mean = 65
n = 50
confidence = 90
alpha = 1 - confidence/100

z = abs(stat.norm.ppf(alpha/2))
margin = z*(std/sqrt(n))
CI = (sample_mean-margin, sample_mean+margin)
print(f"The {confidence}% confidence interval of population mean is {CI}")

The 90% confidence interval of population mean is (63.13906055411732, 66.86093944588268)


Q14. In a study of the effects of caffeine on reaction time, a sample of 30 participants had an average
reaction time of 0.25 seconds with a standard deviation of 0.05 seconds. Conduct a hypothesis test to
determine if the caffeine has a significant effect on reaction time at a 90% confidence level using a t-test.

Ans. We have $\bar{x}=0.25$, $n=30$, $S=0.05$, confidence level = 90%, so $\alpha=1-0.90=0.10$. The null and alternative hypothesis are:
- $H_0$: $\mu=0$ : this means there is no change in reaction time and the caffeine has no effect
- $H_1$: $\mu \ne 0$ : this means there is change in reaction time and the caffeine has effect

As population standard deviation is not given, we will use t-statistics. Using Pyhton, the solution can be obtained as shown:

In [5]:
import scipy.stats as stat
import numpy as np

pop_mean = 0
sample_mean = 0.25
std = 0.05
significance_level = 0.01
sample_size = 30
dof = sample_size-1

t = (sample_mean-pop_mean)/(std/np.sqrt(sample_size))
p = stat.t.ppf(1-significance_level/2, dof)

print("RESULT:")

if t<-p or t>p:
    print("We reject the Null Hypothesis. There is change in reaction time and the caffeine has some effect")
else:
    print("We accept the Null Hypothesis. There is no change in reaction time and the caffeine has no effect")


RESULT:
We reject the Null Hypothesis. There is change in reaction time and the caffeine has some effect
